cluster-stamp.json

```diff
@@ -725,79 +769,47 @@
-                        "query": "let value = toscalar(InsightsMetrics | where Namespace == \"prometheus\" and Name == \"traefik_entrypoint_requests_total\" | where parse_json(Tags).code == 403 | summarize Value = avg(Val) by bin(TimeGenerated, 5m) | summarize min = min(Value)); InsightsMetrics | where Namespace == \"prometheus\" and Name == \"traefik_entrypoint_requests_total\" | where parse_json(Tags).code == 403 | summarize AggregatedValue = avg(Val)-value by bin(TimeGenerated, 5m) | order by TimeGenerated | render barchart",
-                        "version": 1
-                    }
-                },
-                {
-                    "type": "savedSearches",
-                    "apiVersion": "2020-08-01",
-                    "name": "NodeRebootRequested",
-                    "dependsOn": [
-                        "[concat('Microsoft.OperationalInsights/workspaces/', variables('logAnalyticsWorkspaceName'))]"
-                    ],
-                    "properties": {
-                        "eTag": "*",
-                        "category": "Prometheus",
-                        "displayName": "Nodes reboot required by kured",
-                        "query": "InsightsMetrics | where Namespace == \"prometheus\" and Name == \"kured_reboot_required\" | where Val > 0",
-                        "version": 1
-                    }
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "resources": [
+                        {
+                            "type": "Microsoft.Authorization/roleAssignments",
+                            "apiVersion": "2020-04-01-preview",
+                            "name": "[guid(variables('containerRegistryName'), variables('acrPullRole'), variables('clusterControlPlaneIdentityName'))]",
+                            "scope": "[concat('Microsoft.ContainerRegistry/registries/', variables('containerRegistryName'))]",
+                            "properties": {
+                                "roleDefinitionId": "[variables('acrPullRole')]",
+                                "description": "Allows cluster identity to pull container images from this container registry.",
+                                "principalId": "[reference(resourceId('Microsoft.ManagedIdentity/userAssignedIdentities', variables('clusterControlPlaneIdentityName'))).principalId]",
+                                "principalType": "ServicePrincipal"
+                            }
+                        }
+                    ]
                 }
-            ]
+            }
         },
         {
-            "name": "PodFailedScheduledQuery",
+            "name": "[concat('PodFailedScheduledQuery','-',variables('clusterName'))]",
```

Comment:
see if we can use the new way of defining queries and adding scopes later + using the dimension feature. Take a look from regulated.

Reply:
yes, we can and it is now done |  Addressed from [c7bf4](https://github.com/mspnp/aks-secure-baseline/pull/158/commits/c7bf4e10344dd86c5d55eb959538d5388c321d59)